src/adapters/adgear.js

```diff
@@ -0,0 +1,130 @@
+var CONSTANTS = require('../constants.json');
+var utils = require('../utils.js');
+var bidmanager = require('../bidmanager.js');
+var bidfactory = require('../bidfactory.js');
+var adloader = require('../adloader.js');
+var Adapter = require('./adapter.js');
+
+var AdGearAdapter = function AdGearAdapter() {
+  var proto = ('http:' === document.location.protocol) ? 'http' : 'https';
+  var htag = proto + '://rtbx.adgrx.com/impressions/';
+  var baseAdapter = Adapter.createNew('adgear');
+
+  baseAdapter.callBids = function(params) {
+    var bidderCode = baseAdapter.getBidderCode();
+    var bids = [];
+
+    utils._each(params.bids, function(bid) {
+      if (bid.placementCode && bid.sizes && bid.params.tagId) {
+
+        /* We allow overriding 'sizes' via bid.params so check there first: */
+        var sizes = bid.params.hasOwnProperty('sizes') ? bid.params.sizes : bid.sizes;
+        sizes = utils.parseSizesInput(sizes);
+
+        /* We allow publisher overriding 'url' via bid.params so check there first: */
+        var durl = bid.params.hasOwnProperty('url') ? bid.params.url : window.location.href;
+
+        utils._each(sizes, function(format) {
+          var url = htag + bid.params.tagId + '.js?';
+          url = utils.tryAppendQueryString(url, 'AG_CB', 'pbjs.handleAdGearCallback');
+          url = utils.tryAppendQueryString(url, 'AG_SZ', encodeURIComponent(format));
+          url = utils.tryAppendQueryString(url, 'AG_TZ', (new Date()).getTimezoneOffset());
+          url = utils.tryAppendQueryString(url, 'AG_URL', encodeURIComponent(durl));
+          url = utils.tryAppendQueryString(url, 'AG_R', String(Math.floor(Math.random() * 1000000000)));
+
+          var cuid = utils.getUniqueIdentifierStr();
+          url = utils.tryAppendQueryString(url, 'AG_CBID', encodeURIComponent(cuid));
+          bidmanager.pbCallbackMap[cuid] = {
+            'bidObj': bid,
+            'url': url,
+            'start': new Date().getTime()
+          };
+          bids.push(bidmanager.pbCallbackMap[cuid]);
+        });
+
+      }
+    });
+
+    bidmanager.setExpectedBidsCount(bidderCode, bids.length);
```

Comment:
No need for this line as we automatically set bid count to `bids.length` as a default.